### PR TITLE
Add phal adapters

### LINKED ISSUE #65: Cleanup / rewrite switch_configurator.cc
There are several issues from code style to design. Make a pass over the code base and related files.

### PR DESCRIPTION
**Pull request by pudelkoM**
Minimal subset extracted from #43 

Also closes #65

### CODE REVIEW

File: stratum/hal/lib/phal/adapter.cc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+::util::Status Adapter::Subscribe(const std::vector<Path>& paths,
+                                  std::unique_ptr<ChannelWriter<PhalDB>> writer,
+                                  absl::Duration poll_time) {
+  ASSIGN_OR_RETURN(db_query_, attribute_db_interface_->MakeQuery(paths));
+  return db_query_->Subscribe(std::move(writer), poll_time);
+}
+
+::util::Status Adapter::Set(const AttributeValueMap& attrs) {
+  return (attribute_db_interface_->Set(attrs));
```
Comment (Yi-Tseng) [inline stratum/hal/lib/phal/adapter.cc]: `()`

File: .codecov.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ coverage:
   status:
     project:
       default:
-        target: 81%
+        target: 80%
```
Comment (ccascone) [inline .codecov.yml]: Trying comments for issue #69

Comment (pudelkoM) [reply .codecov.yml]: Very good. It even shows up as a "resolvable" comment

File: stratum/hal/lib/phal/onlp/onlpphal.cc [hunk 3]
```diff
@@ -355,10 +293,16 @@ ::util::Status OnlpPhal::InitializeOnlpEventHandler() {
 // Register the configurator so we can use later
 ::util::Status OnlpPhal::RegisterSfpConfigurator(
     int slot, int port, SfpConfigurator* configurator) {
-  const std::pair<int, int> slot_port_pair = std::make_pair(slot, port);
+  const auto slot_port_pair = std::make_pair(slot, port);
+
+  auto onlp_configurator = dynamic_cast<OnlpSfpConfigurator*>(configurator);
+  CHECK_RETURN_IF_FALSE(onlp_configurator)
```
Comment (Yi-Tseng) [inline stratum/hal/lib/phal/onlp/onlpphal.cc]: `onlp_configurator != nullptr` ?

File: stratum/hal/lib/phal/onlp/sfp_configurator_test.cc [hunk 4]
```diff
@@ -152,49 +161,63 @@ class OnlpSfpConfiguratorTest : public ::testing::Test {
...
   OnlpSfpConfigurator* configurator_;
+  onlp_oid_hdr_t mock_oid_info_;
+  onlp_sfp_info_t mock_sfp_info_;
   std::unique_ptr<AttributeDatabase> database_;
   std::unique_ptr<AttributeGroup> root_;
+  std::unique_ptr<SwitchConfigurator> switch_configurator_;
+  SfpConfigurator *sfp_configurator_{nullptr};
   AttributeGroup* sfp_;
-  string phal_config_path_ = "phal_init_config.pb.txt";
+  std::string phal_config_path_ = "phal_init_config.pb.txt";
```
Comment (Yi-Tseng) [inline stratum/hal/lib/phal/onlp/sfp_configurator_test.cc]: unused variable?

File: stratum/hal/lib/phal/onlp/switch_configurator.cc [hunk 5]
```diff
@@ -12,813 +12,658 @@
...
+    mutable_group->AddAttribute(
+        "hardware_state",
+        FixedEnumDataSource::Make(HwState_descriptor(), HW_STATE_FAILED)
+            ->GetAttribute());
+  }
+
+  return result;
 }
 
 ::util::Status OnlpSwitchConfigurator::AddPort(
```
[the remark this answers is not in the document]
Comment (pudelkoM) [reply stratum/hal/lib/phal/onlp/switch_configurator.cc]: yes

File: stratum/hal/config/x86-64-stordis-bf2556x-1t-r0/phal_config.pb.txt [hunk 6]
```diff
@@ -1,283 +1,283 @@
 cards {
-  id: 1
+  slot: 1
   ports {
-    id: 1
+    port: 1
     physical_port_type: PHYSICAL_PORT_TYPE_QSFP_CAGE
```
Comment (stolsma) [inline stratum/hal/config/x86-64-stordis-bf2556x-1t-r0/phal_config.pb.txt]: Are these really QSFP cages? I thought the first 48 ports are SFP+... will check tomorrow....

Comment (pudelkoM) [reply stratum/hal/config/x86-64-stordis-bf2556x-1t-r0/phal_config.pb.txt]: I don't know, but this is not part of this PR. Default configs were merged in #58. Make a new issue if they spot any errors.

Comment (Yi-Tseng) [reply stratum/hal/config/x86-64-stordis-bf2556x-1t-r0/phal_config.pb.txt]: @stolsma You are right, I think I copied the wrong config. I can create another PR to fix that

Thanks

**Pull request merged**